![build workflow](https://github.com/sib-swiss/single-cell-training/actions/workflows/docker-image.yml/badge.svg)
[![License: CC BY-SA 4.0](https://img.shields.io/badge/License-CC_BY--SA_4.0-lightgrey.svg)](https://creativecommons.org/licenses/by-sa/4.0/)

This website is hosted at: https://bioinfocz.github.io/Course-on-scRNA-seq-data-analysis/

Original SIB course is at: https://sib-swiss.github.io/single-cell-training/
# Course website single cell transcriptomics

Single-cell RNA sequencing (scRNA-seq) allows researchers to study gene expression at the level of individual cells. This approach can, for example, help to identify different cell populations in a complex sample and describe their expression patterns. To generate and analyse scRNA-seq data, several methods are available, all with their strengths and weaknesses depending on the researchers’ needs. This 3-day course will cover the main technologies as well as the main aspects to consider while designing an scRNA-seq experiment. In particular, it will combine the theoretical background of analytical methods with hands-on data analysis sessions focused on data generated by droplet-based platforms.


By the end of the course, participants will possess the following abilities:

  - Distinguish advantages and pitfalls of scRNA-seq.
  - Design their own scRNA-seq experiment, using common technologies like 10× Genomics.
  - Apply quality control (QC) measures and utilise analysis tools to preprocess scRNA-seq data.
  - Apply normalisation, scaling, dimensionality reduction, integration and clustering on scRNA-seq data using R.
  - Differentiate between cell annotation techniques to identify and characterise cell populations.
  - Use differential gene expression analysis methods on scRNA-seq data to gain biological insights.
  - Select enrichment analysis methods appropriate to the biological question and data.
  - Develop an scRNA-seq data analysis workflow from raw count matrix to differential gene expression with peer support and light guidance.
    
![project logo](assets/logo_stranky2.png)

## Course organizators and teachers

- Michal Kolar [ORCiD](https://orcid.org/0000-0002-4593-1525)
- Lucie Pfeiferova [ORCiD](https://orcid.org/0000-0003-1089-0329)
- Jan Kubovciak [ORCiD](https://orcid.org/0000-0001-6446-9797)
- Yusuf Caglar Odabasi [ORCiD](https://orcid.org/0000-0003-0690-5575)

## Course support team
- Vojtech Melichar [ORCiD](https://orcid.org/0009-0000-7278-923X)
- Mathys Delattre
- Eva Rohlova

## Atributions
We would like to thank Elixir for all the support and provided computational funds, also we would like to thank Metacentrum organization. This course is largely based on material from the SIB [sib-swiss.github.io/single-cell-r-training](https://sib-swiss.github.io/single-cell-r-training/), which we gratefully acknowledge. Parts of the original SIB Course were inspired by the [Broad Institute Single Cell Workshop](https://broadinstitute.github.io/2019_scWorkshop/index.html), the [CRUK CI Introduction to single-cell RNA-seq data analysis course](https://bioinformatics-core-shared-training.github.io/UnivCambridge_ScRnaSeq_Nov2021/) and courses previously developed by Walid Gharib at SIB. 

### Original authors

- Tania Wyss [ORCiD](https://orcid.org/0000-0003-2641-0895)
- Rachel Marcone-Jeitziner [ORCiD](https://orcid.org/0000-0002-5711-8435)
- Geert van Geest [ORCiD](https://orcid.org/0000-0002-1561-078X)
- Patricia Palagi [ORCiD](https://orcid.org/0000-0001-9062-6303)


## How reuse this material
This website is generated with quarto. To re-build the website, download and install Rstudio and Quarto CLI. Also make sure you have installed the required packages. After that, clone this repository:

```sh
git clone https://github.com/bioinfocz/Course-on-scRNA-seq-data-analysis
```
or for original SIB version

```
git clone https://github.com/sib-swiss/single-cell-r-training.git
```
### Using renv
This project uses renv to manage R dependencies. To restore the project's environment, run the following commands in R:

```R
install.packages("renv")
renv::restore()
```
### Render website
Open the project in Rstudio, and run in the terminal to render the full website:

```sh
quarto render
```
Sites are rendered to docs folder. 

### Run with Docker
A Docker image with all the required software is available on Docker Hub.

To run the Docker container, you can use the following command:

```sh
docker run \
--rm \
-p 8787:8787 \
-v $PWD:/home/rstudio \
sibswiss/training-singlecell-rstudio:latest
```
You can also use the script `Docker/run_locally.sh` to run the container.

### Update Docker image
To add or update R packages in the Docker image, you need to:

1. Add the new packages to the `Docker/install_packages.R` script.
2. Copy the `renv.lock` file from the root directory to the `Docker` directory.
3. Rebuild the Docker image.

#### Using GitHub Actions
You can manually trigger a build of the Docker image by going to the Actions tab of this repository, selecting the "Manual build and push" workflow, and clicking "Run workflow".

#### Locally
Alternatively, you can rebuild the image locally:

```sh
cp renv.lock Docker/
cd Docker
docker build -t sibswiss/training-singlecell-rstudio:latest .
```
